File: Godot-MCP/server/src/llm-connectors/groq_client.ts
import Groq from 'groq-sdk';
import { BaseLlmClient } from './base_client';

export class GroqClient implements BaseLlmClient {
    private readonly client: Groq;
    private readonly model: string;

    constructor() {
        const apiKey = process.env.GROQ_API_KEY;
        this.model = process.env.GROQ_MODEL || 'llama3-70b-8192';

        if (!apiKey || apiKey.includes('YOUR_GROQ_API_KEY_HERE')) {
            throw new Error('Groq API key is missing or not set in .env file');
        }
        this.client = new Groq({ apiKey });
    }

    async generatePlan(systemPrompt: string, userPrompt: string): Promise<string> {
        const response = await this.client.chat.completions.create({
            model: this.model,
            messages: [
                { role: "system", content: systemPrompt },
                { role: "user", content: userPrompt },
            ],
            response_format: { type: "json_object" },
        });
        return response.choices[0].message.content || '';
    }
}

File: bmad_method/llm_connectors/groq_client.py
import os
from groq import Groq
from .base_client import BaseLlmClient

class GroqClient(BaseLlmClient):
    def __init__(self):
        super().__init__()
        self.api_key = os.getenv('GROQ_API_KEY')
        self.model = os.getenv('GROQ_MODEL')
        if not self.api_key or 'YOUR_GROQ_API_KEY_HERE' in self.api_key:
            raise ValueError("Groq API key is missing or not set in .env file")
        self.client = Groq(api_key=self.api_key)

    def generate_plan(self, system_prompt: str, user_prompt: str) -> str:
        response = self.client.chat.completions.create(
            model=self.model,
            messages=[
                {"role": "system", "content": system_prompt},
                {"role": "user", "content": user_prompt}
            ],
            response_format={"type": "json_object"}
        )
        return response.choices[0].message.content
